packages/amplify-e2e-tests/src/__tests__/api_8.test.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  addApiWithoutSchema,
+  addFeatureFlag,
+  addFunction,
+  addRestApi,
+  amplifyPushGraphQlWithCognitoPrompt,
+  amplifyPushUpdate,
+  checkIfBucketExists,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  getAppSyncApi,
+  getProjectMeta,
+  initJSProjectWithProfile,
+  updateApiSchema,
+} from 'amplify-e2e-core';
+import { readdirSync, readFileSync } from 'fs';
+import * as path from 'path';
+
+// to deal with bug in cognito-identity-js
+(global as any).fetch = require('node-fetch');
+// to deal with subscriptions in node env
+(global as any).WebSocket = require('ws');
+
+describe('amplify add api (REST)', () => {
+  let projRoot: string;
+  beforeEach(async () => {
+    projRoot = await createNewProjectDir('rest-api');
+  });
+
+  afterEach(async () => {
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot);
+  });
+
+  const validateMeta = async (meta?) => {
+    meta = meta ?? getProjectMeta(projRoot);
+    expect(meta.providers.awscloudformation).toBeDefined();
+    const {
+      AuthRoleArn: authRoleArn,
+      UnauthRoleArn: unauthRoleArn,
+      DeploymentBucketName: bucketName,
+      Region: region,
+      StackId: stackId,
+    } = meta.providers.awscloudformation;
+    expect(authRoleArn).toBeDefined();
+    expect(unauthRoleArn).toBeDefined();
+    expect(region).toBeDefined();
+    expect(stackId).toBeDefined();
+    const bucketExists = await checkIfBucketExists(bucketName, region);
+    expect(bucketExists).toMatchObject({});
+
+    expect(meta.function).toBeDefined();
+    let seenAtLeastOneFunc = false;
+    for (let key of Object.keys(meta.function)) {
+      const { service, build, lastBuildTimeStamp, lastPackageTimeStamp, distZipFilename, lastPushTimeStamp, lastPushDirHash } =
+        meta.function[key];
+      expect(service).toBe('Lambda');
+      expect(build).toBeTruthy();
+      expect(lastBuildTimeStamp).toBeDefined();
+      expect(lastPackageTimeStamp).toBeDefined();
+      expect(distZipFilename).toBeDefined();
+      expect(lastPushTimeStamp).toBeDefined();
+      expect(lastPushDirHash).toBeDefined();
+      seenAtLeastOneFunc = true;
+    }
+    expect(seenAtLeastOneFunc).toBe(true);
+  };
```

Comment:
If this code is shard, can it be moved to a common file instead of having multiple copies

Reply:
Moved to a common file under amplify-e2e-core package.